Submitted By: DJ Lucas <dj AT linuxfromscratch DOT org>
Date: 2012-05-26
Initial Package Version: 1.9
Upstream Status: Not applicable
Origin: http://icedtea.classpath.org/hg/icedtea6/
Description: Includes various fixes from upstream to allow a for a proper
             bootstrap build.

diff -Naurp icedtea-2.2-orig/Makefile.am icedtea-2.2/Makefile.am
--- icedtea-2.2-orig/Makefile.am	2012-05-26 09:24:34.000000000 -0500
+++ icedtea-2.2/Makefile.am	2012-05-26 09:28:39.000000000 -0500
@@ -280,6 +280,8 @@ if WITH_PAX
 ICEDTEA_PATCHES += patches/test_gamma.patch
 endif
 
+DISTRIBUTION_PATCHES = patches/lfs-fixed-paths.patch
+
 ICEDTEA_PATCHES += $(DISTRIBUTION_PATCHES)
 
 # Bootstrapping patches
@@ -746,7 +748,8 @@ clean-local: clean-tests clean-jtreg-rep
  clean-bootstrap-directory-symlink-stage1 clean-bootstrap-directory-symlink-stage2 \
  clean-extract clean-generated clean-native-ecj clean-hgforest clean-icedtea-stage2 \
  clean-icedtea-debug-stage2 clean-icedtea-stage1 clean-add-zero clean-add-zero-debug \
- clean-add-cacao clean-add-cacao-debug clean-rt clean-rewrite-rhino clean-rewriter
+ clean-add-cacao clean-add-cacao-debug clean-rt clean-rewrite-rhino clean-rewriter \
+ clean-cacerts clean-cacerts-debug
 	if [ -e bootstrap ]; then \
 	  rmdir bootstrap ; \
 	fi
@@ -776,7 +779,7 @@ install:
 	jtregcheck clean-remove-intree-libraries \
 	clean-jamvm clean-extract-jamvm clean-add-jamvm clean-add-jamvm-debug \
 	clean-extract-hotspot clean-sanitise-openjdk \
-	clean-tests clean-tapset-report
+	clean-tests clean-tapset-report clean-cacerts clean-cacerts-debug
 
 env:
 	@echo 'unset JAVA_HOME'
@@ -1703,6 +1706,20 @@ endif
 if ENABLE_JAMVM
 	printf -- '-jamvm ALIASED_TO -server\n' >> $(BUILD_JRE_ARCH_DIR)/jvm.cfg
 endif
+if GENERATE_CACERTS
+	$(ARCH_PREFIX) $(MAKE) add-cacerts
+else
+if CACERTS_FILE_SET
+	$(ARCH_PREFIX) $(MAKE) add-cacerts
+endif
+endif
+if GENERATE_CACERTS
+	$(ARCH_PREFIX) $(MAKE) add-cacerts
+else
+if CACERTS_FILE_SET
+	$(ARCH_PREFIX) $(MAKE) add-cacerts
+endif
+endif
 	@echo "IcedTea is served:" $(BUILD_OUTPUT_DIR)
 	mkdir -p stamps
 	touch $@
@@ -1769,6 +1786,20 @@ endif
 if ENABLE_JAMVM
 	printf -- '-jamvm ALIASED_TO -server\n' >> $(BUILD_DEBUG_JRE_ARCH_DIR)/jvm.cfg
 endif
+if GENERATE_CACERTS
+	$(ARCH_PREFIX) $(MAKE) add-cacerts-debug
+else
+if CACERTS_FILE_SET
+	$(ARCH_PREFIX) $(MAKE) add-cacerts-debug
+endif
+endif
+if GENERATE_CACERTS
+	$(ARCH_PREFIX) $(MAKE) add-cacerts-debug
+else
+if CACERTS_FILE_SET
+	$(ARCH_PREFIX) $(MAKE) add-cacerts-debug
+endif
+endif
 	@echo "IcedTea (debug build) is served:" \
 	  $(DEBUG_BUILD_OUTPUT_DIR)
 	mkdir -p stamps
@@ -1857,6 +1888,132 @@ stamps/icedtea-stage1.stamp: stamps/iced
 clean-icedtea-stage1:
 	rm -f stamps/icedtea-stage1.stamp
 
+# CA Certs
+stamps/generate-cacerts.stamp:
+if GENERATE_CACERTS
+	if test -n "${CADIR}"; then \
+	  sh scripts/mkcacerts.sh -d "${CADIR}" \
+	    -k $(BUILD_OUTPUT_DIR)/j2sdk-image/bin/keytool \
+	    -s $(OPENSSL) \
+	    -o $(BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts; \
+	else \
+	  sh scripts/mkcacerts.sh -f "${CAFILE}" \
+	    -k $(BUILD_OUTPUT_DIR)/j2sdk-image/bin/keytool \
+	    -s $(OPENSSL) \
+	    -o $(BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts; \
+	fi
+endif
+	touch stamps/generate-cacerts.stamp
+
+stamps/add-cacerts.stamp: stamps/generate-cacerts.stamp
+if CACERTS_FILE_SET
+	cp $(CACERTS_FILE) $(BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts
+endif
+	cp $(BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts \
+	   $(BUILD_OUTPUT_DIR)/j2sdk-image/jre/lib/security/cacerts
+	touch stamps/add-cacerts.stamp
+
+clean-cacerts:
+	rm -f $(BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts
+	rm -f $(BUILD_OUTPUT_DIR)/j2sdk-image/jre/lib/security/cacerts
+	rm -f stamps/add-cacerts.stamp
+	rm -f stamps/generate-cacerts.stamp
+
+stamps/generate-cacerts-debug.stamp:
+if GENERATE_CACERTS
+	if test -n "${CADIR}"; then \
+	  sh scripts/mkcacerts.sh -d "${CADIR}" \
+	    -k $(DEBUG_BUILD_OUTPUT_DIR)/j2sdk-image/bin/keytool \
+	    -s $(OPENSSL) \
+	    -o $(DEBUG_BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts; \
+	else \
+	  sh scripts/mkcacerts.sh -f "${CAFILE}" \
+	    -k $(DEBUG_BUILD_OUTPUT_DIR)/j2sdk-image/bin/keytool \
+	    -s $(OPENSSL) \
+	    -o $(DEBUG_BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts; \
+	fi
+endif
+	touch stamps/generate-cacerts-debug.stamp
+
+stamps/add-cacerts-debug.stamp: stamps/generate-cacerts-debug.stamp
+if CACERTS_FILE_SET
+	cp $(CACERTS_FILE) $(DEBUG_BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts
+endif
+	cp $(DEBUG_BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts \
+	   $(DEBUG_BUILD_OUTPUT_DIR)/j2sdk-image/jre/lib/security/cacerts
+	touch stamps/add-cacerts-debug.stamp
+
+clean-cacerts-debug:
+	rm -f $(DEBUG_BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts
+	rm -f $(DEBUG_BUILD_OUTPUT_DIR)/j2sdk-image/jre/lib/security/cacerts
+	rm -f stamps/add-cacerts-debug.stamp
+	rm -f stamps/generate-cacerts-debug.stamp
+
+# end of CA Certs
+
+# CA Certs
+stamps/generate-cacerts.stamp:
+if GENERATE_CACERTS
+	if test -n "${CADIR}"; then \
+	  sh scripts/mkcacerts.sh -d "${CADIR}" \
+	    -k $(BUILD_OUTPUT_DIR)/j2sdk-image/bin/keytool \
+	    -s $(OPENSSL) \
+	    -o $(BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts; \
+	else \
+	  sh scripts/mkcacerts.sh -f "${CAFILE}" \
+	    -k $(BUILD_OUTPUT_DIR)/j2sdk-image/bin/keytool \
+	    -s $(OPENSSL) \
+	    -o $(BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts; \
+	fi
+endif
+	touch stamps/generate-cacerts.stamp
+
+stamps/add-cacerts.stamp: stamps/generate-cacerts.stamp
+if CACERTS_FILE_SET
+	cp $(CACERTS_FILE) $(BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts
+endif
+	cp $(BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts \
+	   $(BUILD_OUTPUT_DIR)/j2sdk-image/jre/lib/security/cacerts
+	touch stamps/add-cacerts.stamp
+
+clean-cacerts:
+	rm -f $(BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts
+	rm -f $(BUILD_OUTPUT_DIR)/j2sdk-image/jre/lib/security/cacerts
+	rm -f stamps/add-cacerts.stamp
+	rm -f stamps/generate-cacerts.stamp
+
+stamps/generate-cacerts-debug.stamp:
+if GENERATE_CACERTS
+	if test -n "${CADIR}"; then \
+	  sh scripts/mkcacerts.sh -d "${CADIR}" \
+	    -k $(DEBUG_BUILD_OUTPUT_DIR)/j2sdk-image/bin/keytool \
+	    -s $(OPENSSL) \
+	    -o $(DEBUG_BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts; \
+	else \
+	  sh scripts/mkcacerts.sh -f "${CAFILE}" \
+	    -k $(DEBUG_BUILD_OUTPUT_DIR)/j2sdk-image/bin/keytool \
+	    -s $(OPENSSL) \
+	    -o $(DEBUG_BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts; \
+	fi
+endif
+	touch stamps/generate-cacerts-debug.stamp
+
+stamps/add-cacerts-debug.stamp: stamps/generate-cacerts-debug.stamp
+if CACERTS_FILE_SET
+	cp $(CACERTS_FILE) $(DEBUG_BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts
+endif
+	cp $(DEBUG_BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts \
+	   $(DEBUG_BUILD_OUTPUT_DIR)/j2sdk-image/jre/lib/security/cacerts
+	touch stamps/add-cacerts-debug.stamp
+
+clean-cacerts-debug:
+	rm -f $(DEBUG_BUILD_OUTPUT_DIR)/j2re-image/lib/security/cacerts
+	rm -f $(DEBUG_BUILD_OUTPUT_DIR)/j2sdk-image/jre/lib/security/cacerts
+	rm -f stamps/add-cacerts-debug.stamp
+	rm -f stamps/generate-cacerts-debug.stamp
+
+# end of CA Certs
+
 # PulseAudio based mixer
 # (pulse-java)
 if ENABLE_PULSE_JAVA
@@ -2374,6 +2531,14 @@ clean-rt:
 # Target Aliases
 # ===============
 
+add-cacerts: stamps/add-cacerts.stamp
+
+add-cacerts-debug: stamps/add-cacerts-debug.stamp
+
+add-cacerts: stamps/add-cacerts.stamp
+
+add-cacerts-debug: stamps/add-cacerts-debug.stamp
+
 add-zero: stamps/add-zero.stamp
 
 add-zero-debug: stamps/add-zero-debug.stamp
diff -Naurp icedtea-2.2-orig/patches/lfs-fixed-paths.patch icedtea-2.2/patches/lfs-fixed-paths.patch
--- icedtea-2.2-orig/patches/lfs-fixed-paths.patch	1969-12-31 18:00:00.000000000 -0600
+++ icedtea-2.2/patches/lfs-fixed-paths.patch	2012-05-26 09:28:49.000000000 -0500
@@ -0,0 +1,55 @@
+diff -Nru openjdk.orig/corba/make/common/shared/Defs-utils.gmk openjdk/corba/make/common/shared/Defs-utils.gmk
+--- openjdk-orig/corba/make/common/shared/Defs-utils.gmk	2010-06-21 16:11:20.000000000 -0500
++++ openjdk/corba/make/common/shared/Defs-utils.gmk	2011-01-11 01:30:48.000000000 -0600
+@@ -88,12 +88,12 @@
+ EGREP          = $(UTILS_COMMAND_PATH)egrep
+ EXPR           = $(UTILS_USR_BIN_PATH)expr
+ FILE           = $(UTILS_USR_BIN_PATH)file
+-FIND           = $(UTILS_USR_BIN_PATH)find
++FIND           = $(UTILS_COMMAND_PATH)find
+ FMT            = $(UTILS_COMMAND_PATH)fmt
+ GDB            = $(UTILS_USR_BIN_PATH)gdb
+ GREP           = $(UTILS_COMMAND_PATH)grep
+ GUNZIP         = $(UTILS_COMMAND_PATH)gunzip
+-HEAD           = $(UTILS_USR_BIN_PATH)head
++HEAD           = $(UTILS_COMMAND_PATH)head
+ ID             = $(UTILS_COMMAND_PATH)id
+ ISAINFO        = $(UTILS_COMMAND_PATH)isainfo
+ KSH            = $(UTILS_COMMAND_PATH)ksh
+@@ -125,7 +125,7 @@
+ TAIL           = $(UTILS_USR_BIN_PATH)tail
+ TAR            = $(UTILS_COMMAND_PATH)tar
+ TEST           = $(UTILS_USR_BIN_PATH)test
+-TOUCH          = $(UTILS_COMMAND_PATH)touch
++TOUCH          = $(UTILS_USR_BIN_PATH)touch
+ TR             = $(UTILS_USR_BIN_PATH)tr
+ TRUE           = $(UTILS_COMMAND_PATH)true
+ UNAME          = $(UTILS_COMMAND_PATH)uname
+--- openjdk-orig/jdk/make/common/shared/Defs-utils.gmk	2012-04-16 02:09:00.000000000 +0000
++++ openjdk/jdk/make/common/shared/Defs-utils.gmk	2012-04-16 02:10:44.000000000 +0000
+@@ -104,14 +104,14 @@
+ EGREP          = $(UTILS_COMMAND_PATH)egrep
+ EXPR           = $(UTILS_USR_BIN_PATH)expr
+ FILE           = $(UTILS_USR_BIN_PATH)file
+-FIND           = $(UTILS_USR_BIN_PATH)find
++FIND           = $(UTILS_COMMAND_PATH)find
+ FMT            = $(UTILS_COMMAND_PATH)fmt
+ GDB            = $(UTILS_USR_BIN_PATH)gdb
+ GREP           = $(UTILS_COMMAND_PATH)grep
+ GUNZIP         = $(UTILS_COMMAND_PATH)gunzip
+ # GZIP is used for solaris. Linux and windows use tar czf
+ GZIP           = $(UTILS_COMMAND_PATH)gzip
+-HEAD           = $(UTILS_USR_BIN_PATH)head
++HEAD           = $(UTILS_COMMAND_PATH)head
+ HG             = hg
+ ID             = $(UTILS_COMMAND_PATH)id
+ ISAINFO        = $(UTILS_COMMAND_PATH)isainfo
+@@ -140,7 +140,7 @@
+ TAIL           = $(UTILS_USR_BIN_PATH)tail
+ TAR            = $(UTILS_COMMAND_PATH)tar
+ TEST           = $(UTILS_USR_BIN_PATH)test
+-TOUCH          = $(UTILS_COMMAND_PATH)touch
++TOUCH          = $(UTILS_USR_BIN_PATH)touch
+ TR             = $(UTILS_USR_BIN_PATH)tr
+ TRUE           = $(UTILS_COMMAND_PATH)true
+ UNAME          = $(UTILS_COMMAND_PATH)uname
